test/one/profiler/test/TestProcess.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.profiler.test;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.io.Closeable;
+import java.io.*;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import one.jfr.JfrReader;
+
+public class TestProcess implements Closeable {
+    private static final Logger log = Logger.getLogger(TestProcess.class.getName());
+
+    private static final Pattern filePattern = Pattern.compile("(%[a-z]+)(\\.[a-z]+)?");
+
+    private static final MethodHandle pid = getPidHandle();
+
+    private static MethodHandle getPidHandle() {
+        // JDK 9+
+        try {
+            return MethodHandles.publicLookup().findVirtual(Process.class, "pid", MethodType.methodType(long.class));
+        } catch (ReflectiveOperationException e) {
+            // fallback
+        }
+
+        // JDK 8
+        try {
+            Field f = Class.forName("java.lang.UNIXProcess").getDeclaredField("pid");
+            f.setAccessible(true);
+            return MethodHandles.lookup().unreflectGetter(f).asType(MethodType.methodType(long.class, Process.class));
+        } catch (ReflectiveOperationException e) {
+            throw new IllegalStateException("Unsupported API", e);
+        }
+    }
+
+    private final Process p;
+    private final Map<String, File> tmpFiles = new HashMap<>();
+    private final int timeout = 30;
+    private String testName = null;
+
+    public TestProcess(Test test, String libExt, String testName) throws Exception {
+        this(test, libExt);
+        this.testName = testName; // if testName is not null then we retain the logs into that folder
```

Comment:
It's better to avoid such implicit dependencies

Reply:
Removed. Logs will always be retained